packages/flutter/lib/src/widgets/gesture_detector.dart

```diff
@@ -150,6 +150,55 @@ class GestureRecognizerFactoryWithHandlers<T extends GestureRecognizer> extends
 /// ** See code in examples/api/lib/widgets/gesture_detector/gesture_detector.1.dart **
 /// {@end-tool}
 ///
+/// ## Troubleshooting
+///
+/// Why isn't my parent [GestureDetector] onTap method called?
+///
+/// Given a parent [GestureDetector] which defines an onTap callback and a child
+/// [GestureDetector] that also defines an onTap callback. When the inner
+/// [GestureDetector] is tapped, both [GestureDetector]s send a [GestureRecognizer]
+/// into the gesture arena because the pointer coordinates are in both
+/// [GestureDetector]s bounds. The child [GestureDetector] wins because it was
+/// the first detector to enter the arena and it got first dips because it is in front.
+/// For more information on gesture desambiguation see:
+/// [Gesture disambiguation](https://docs.flutter.dev/development/ui/advanced/gestures#gesture-disambiguation)
+///
+/// Setting [GestureDetector.behavior] to [HitTestBehavior.opaque]
+/// or [HitTestBehavior.translucent] has no impact on parent-child relationships
+/// like in this case. To block hits on children use an [IgnorePointer] or [AbsorbPointer].
+///
+/// Some callbacks are called during the gesture arena resolution
+/// (for instance onTapDown). If the parent [GestureDetector] defines such callbacks,
+/// they will be called. Some callbacks are conditionnaly called, for instance
+/// [TapGestureRecognizer.onTapDown] triggers once a short timeout has elapsed.
+///
+/// {@tool dartpad}
+/// This example uses a [GestureDetector] that wraps a green [Container] and a second
+/// [GestureDetector] that wraps a yellow [Container]. The second [GestureDetector] is
+/// a child of the green [Container].
+/// Both [GestureDetector]s define an onTap callback. When the callback is called it
+/// adds a red border to the corresponding [Container].
+///
+/// When the green [Container] is tapped, it's parent [GestureDetector] enters
+/// the gesture arena. It wins because there is no competing [GestureDetector] and
+/// the green [Container] shows a red border.
+/// When the yellow [Container] is tapped, it's parent [GestureDetector] enters
+/// the gesture arena. The [GestureDetector] that wraps the green [Container] also
+/// enters the gesture arena (the pointer events coordinates are inside both
+/// [GestureDetector]s bounds). The [GestureDetector] that wraps the yellow [Container]
+/// wins because it was the first detector to enter the arena and it got first dips
+/// because it is in front.
+///
+/// This example sets [debugPrintGestureArenaDiagnostics] to true.
```

Comment:
looks like this is actually commented out in the example?

Reply:
@goderbauer I commented out this line after pushing the PR because an integration test failed. This integration test runs all code examples and checks that no `debug` constants are updated. I will have a second look at this when I will push my next commit, if I remember well, I chose to comment this line instead of complexifying the code sample. I agree it would be better to comment out this line.

Reply:
I found a way to reset `debugPrintGestureArenaDiagnostics` without over-complicating the example code (using the dispose method of the state class)